### PR TITLE
Bump dependencies (with tests) + Image Optimization

### PR DESCRIPTION
**Pull request by szb512**
Tests have been run and all pull requests have been rebased and merged.

### CODE REVIEW

**Comment by szb512** [comment]
I put tests on all pull requests.

File: package.json [hunk 1]
```diff
@@ -34,79 +34,80 @@
...
-    "nock": "^9.2.3",
+    "nock": "^10.0.6",
     "node-gyp": "^3.8.0",
     "node-sass": "^4.7.2",
     "pkg": "^4.3.0",
     "sass-lint": "^1.12.1",
     "simple-progress-webpack-plugin": "^1.1.2",
     "spectron": "^5.0.0",
-    "webpack": "^4.27.0",
+    "versionist": "^4.1.1",
```
Comment (lurch) [inline package.json]: You seem to be re-adding `versionist` when #2692 removed it.

Comment (szb512) [reply package.json]: True. I'll remove it.

**Comment by thundron** [comment]
This looks a bit confusing for now, I'll have a look at it when I get the chance but dependencies are generally something we treat very carefully due to the shrinkwrap being constantly changed along with them, so we update versions just if we *really* need to (also way too many "Merge ..." commits in there!)
Image optimization looks nice though 👍 you could maybe split that in a separate PR and clean this one up a bit

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-codespell==1.14.0
+codespell==1.15.0
 cpplint==1.4.4
-awscli==1.16.135 
+awscli==1.16.16
```
Comment (lurch) [inline requirements.txt]: Typo? https://pypi.org/project/awscli/ says the latest version is `1.16.161`

Comment (szb512) [reply requirements.txt]: I used github again.

Comment (szb512) [reply requirements.txt]: I'll change it.

Comment (szb512) [reply requirements.txt]: Fixed.

Comment (lurch) [reply requirements.txt]: https://pypi.org/project/awscli/ now says `awscli 1.16.176` :wink:

Comment (szb512) [reply requirements.txt]: ok

**Pull request closed without merging**